programs/uxd/tests/integration_tests/api/program_uxd/instructions/process_redeem_from_mercurial_vault_depository.rs

```diff
@@ -0,0 +1,187 @@
+        MercurialVaultDepository,
+    >(program_test_context, &mercurial_vault_depository)
+    .await?;
+
+    let user_collateral_amount_before =
+        program_test_context::read_account_packed::<Account>(program_test_context, user_collateral)
+            .await?
+            .amount;
+    let user_redeemable_amount_before =
+        program_test_context::read_account_packed::<Account>(program_test_context, user_redeemable)
+            .await?
+            .amount;
+
+    // Execute IX
+    let accounts = uxd::accounts::RedeemFromMercurialVaultDepository {
+        payer: payer.pubkey(),
+        user: user.pubkey(),
+        controller,
+        collateral_mint: *collateral_mint,
+        redeemable_mint,
+        user_collateral: *user_collateral,
+        user_redeemable: *user_redeemable,
+        depository: mercurial_vault_depository,
+        depository_lp_token_vault: mercurial_vault_depository_lp_token_vault,
+        mercurial_vault,
+        mercurial_vault_lp_mint: *mercurial_vault_lp_mint,
+        mercurial_vault_collateral_token_safe,
+        system_program: anchor_lang::system_program::ID,
+        token_program: anchor_spl::token::ID,
+        mercurial_vault_program: mercurial_vault::ID,
+    };
+    let payload = uxd::instruction::RedeemFromMercurialVaultDepository { redeemable_amount };
+    let instruction = Instruction {
+        program_id: uxd::id(),
+        accounts: accounts.to_account_metas(None),
+        data: payload.data(),
+    };
+    program_test_context::process_instruction_with_signer(
+        program_test_context,
+        instruction,
+        payer,
+        user,
+    )
+    .await?;
+
+    // Read state after
+    let redeemable_mint_after =
+        program_test_context::read_account_packed::<Mint>(program_test_context, &redeemable_mint)
+            .await?;
+    let controller_after =
+        program_test_context::read_account_anchor::<Controller>(program_test_context, &controller)
+            .await?;
+    let mercurial_vault_depository_after = program_test_context::read_account_anchor::<
+        MercurialVaultDepository,
+    >(program_test_context, &mercurial_vault_depository)
+    .await?;
+
+    let user_collateral_amount_after =
+        program_test_context::read_account_packed::<Account>(program_test_context, user_collateral)
+            .await?
+            .amount;
+    let user_redeemable_amount_after =
+        program_test_context::read_account_packed::<Account>(program_test_context, user_redeemable)
+            .await?
+            .amount;
+
+    // Compute expected collateral amount after minting fees
+    let collateral_amount = calculate_amount_less_fees(
+        redeemable_amount,
+        mercurial_vault_depository_before.minting_fee_in_bps,
+    )
+    .map_err(program_test_context::ProgramTestError::Anchor)?;
+    let fees_amount = redeemable_amount - collateral_amount;
+
+    // Check result
+    let redeemable_mint_supply_before = redeemable_mint_before.supply;
```

Comment:
This is not optimal readbility wise but there was the same issue in TS. Don't have a solution I guess it's just inherent to complex things to test

Reply:
Yeah for some reason rust force us to create local variable for each of those too (unalign-struct error otherwise)

I could separate each assert into its own chunk of code, or even functions? not sure if that would improve readability, what do you think?